### PR TITLE
Add a native JSON data type support in MySQL

### PR DESCRIPTION
**Pull request by kamipo**
As of MySQL 5.7.8, MySQL supports a native JSON data type.

http://dev.mysql.com/doc/refman/5.7/en/json.html

Example:

``` ruby
create_table :json_data_type do |t|
  t.json :settings
end
```

### CODE REVIEW

**Comment by jeremy** [comment]
:+1: nice @kamipo !

**Comment by sahin** [comment]
+1

File: activerecord/lib/active_record/connection_adapters/postgresql/oid/json.rb [hunk 1]
```diff
@@ -2,32 +2,7 @@ module ActiveRecord
   module ConnectionAdapters
```
Comment (rafaelfranca) [inline activerecord/lib/active_record/connection_adapters/postgresql/oid/json.rb]: We can remove this whole file and use `Type::Json` where we used `OID::JSON`

**Comment by rafaelfranca** [comment]
Awesome! Just remove the OID class and we can merge it.

File: activerecord/lib/active_record/connection_adapters/postgresql/oid/json.rb (hunk 1 above)
Comment (kamipo) [inline activerecord/lib/active_record/connection_adapters/postgresql/oid/json.rb]: `OID::Json` is base class of `OID::Jsonb`.
https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/postgresql/oid/jsonb.rb#L5
Do I need to change to `class Jsonb < Type::Json`?

Comment (rafaelfranca) [inline activerecord/lib/active_record/connection_adapters/postgresql/oid/json.rb]: Yeah, looks good.

**Comment by kamipo** [comment]
Removed `OID::Json`!

**Comment by NeMO84** [comment]
Is this feature only available in the 5.0 version?

**Comment by sgrif** [comment]
Yes

**Comment by akshah123** [comment]
@sgrif are there any plans to add JSON support in 4.x branch?

**Comment by rafaelfranca** [comment]
No. Like stated in our maintenance policy, new features are no applied to
stable releases.

**Pull request merged**